File: vxpt-bbs/vxpt-bbs-server/src/main/java/com/github/blanexie/vxpt/bbs/util/GlobalExceptionHandler.java
package com.github.blanexie.vxpt.bbs.util;

import cn.dev33.satoken.util.SaResult;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler {
    // 全局异常拦截
    @ExceptionHandler
    public SaResult handlerException(Exception e) {
        log.error("错误信息", e);
        return SaResult.error(e.getMessage());
    }
}

File: vxpt-bbs/src/main/java/com/github/blanexie/vxpt/bbs/util/spring/GlobalExceptionHandler.kt
package com.github.blanexie.vxpt.bbs.util.spring

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import javax.servlet.http.HttpServletRequest


@ControllerAdvice
@ResponseBody
class GlobalExceptionHandler {

    val logger: Logger = LoggerFactory.getLogger(GlobalExceptionHandler::class.java)

    @ExceptionHandler(value = [Exception::class])
    fun exceptionHandler(request: HttpServletRequest, e: Exception): WebResp {
        if (e is VxptException) {
            logger.error("全局异常处理器拦截了", e)
            return WebResp.fail(e.respCode.code, e.respCode.message)
        }
        logger.error("全局异常处理器拦截了", e)
        return WebResp.fail(RespCode.SERVER_ERROR.code, e.message ?: "")
    }
}
